service/matching/task_queue_partition_manager.go

```diff
@@ -299,6 +299,12 @@ func (pm *taskQueuePartitionManagerImpl) PollTask(
 				versionSetUsed = true
 				dbq, err = pm.getVersionedQueue(ctx, versionSet, "", true)
 			} else {
+				activeRules := getActiveRedirectRules(versioningData.RedirectRules)
+				terminalBuildId := findTerminalBuildId(buildId, activeRules)
+				if terminalBuildId != buildId {
+					return nil, false, serviceerror.NewNewerBuildExists(terminalBuildId)
```

Comment:
oh, I thought we weren't going to do this. it's good, though, I think we should.